docs/releasing.md

```diff
@@ -1,96 +1,184 @@
 # Release Instructions
 
-Allow 1h to do a release.
+## Release
 
-## Preparation
+### 1. Cherry-pick Issue
 
-* [ ] Cherry-pick your changes from master onto the release branch.
-* [ ] The release branch should be green in CI before you start.
+Create a cherry-pick issue to allow the team and community to comment on the release contents.
 
-## Release
+1. Locate the previous cherry-pick issue
+2. Get the hash of the most recent commit still available on the previous issue
+3. Generate new issue contents:
+    
+    ```sh
+    $ git checkout master # Ensure we are on master
+    $ git log --pretty=format:"%an: %s %h"  [COMMIT_HASH]..HEAD
+    ```
+4. Create a new issue on GitHub with the title `[VERSION] cherry-pick` (e.g. `v3.0.2 cherry-pick`) and the generated commits
+as content.
 
-Releasing requires a clean tree state, so back-up any untracked files in your Git directory.
+### 2. Cherry-pick to Release Branch
 
-**Only once your files are backed up**, run:
+Once the team and community is satisfied with the commits to be cherry-picked, cherry-pick them into the appropriate
+release branch. There should be a single release branch per minor release (e.g. `release-3.0`, `release-3.1`, etc.)
 
-    git clean -fdx  # WARNING: Will delete untracked files!
+1. Checkout the release branch and cherry-pick commits
 
-To generate new manifests and perform basic checks:
+    ```sh
+    $ git checkout relesae-3.0
+    $ git cherry-pick [COMMIT_IDS...]
+    ```
 
-    make prepare-release -B VERSION=v2.11.5
+2. Hope for few merge conflicts!
 
```

Comment:
I think it's worth mentioning how to handle merge conflicts in this scenario? This is just my 2 cents of how to handle cherry-picking conflicts. Hopefully it's useful!

```suggestion
2. Hope for few merge conflicts!

    A merge conflict during cherry-picking usually means the commit is based on another commit that should be 
    cherry-picked first. In case of a merge conflict, you can undo the cherry-picking by `git cherry-pick --abort` and 
    revisit the list of commits to make sure the prior commits are cherry-picked as well.

```